url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/Console.java

```diff
@@ -0,0 +1,623 @@
+
+            String inputFilename = options.get(KW_INPUT_FILE);
+            if (inputFilename == null) {
+                throw new RuntimeException("[option error] no input file specified");
+            }
+            File inputFile = new File(inputFilename);
+            if (!inputFile.exists()) {
+                inputFile.createNewFile();
+            }
+
+            BufferedReader reader = new BufferedReader(new FileReader(inputFile));
+            try {
+                String line = reader.readLine();
+                while (line != null) {
+                    textEngine.append(line).append("\n");
+                    line = reader.readLine();
+                }
+            } catch (Exception ex) {
+                throw new RuntimeException(ex);
+            } finally {
+                reader.close();
+            }
+
+            inputFormat = options.get(KW_INPUT_FORMAT);
+            if (inputFormat == null || inputFormat.isEmpty()) {
+                int extensionIndex = inputFilename.lastIndexOf(".");
+                if (extensionIndex >= 0) {
+                    inputFormat = inputFilename.substring(extensionIndex + 1);
+                } else {
+                    throw new RuntimeException("[format error] unspecified format of input file");
+                }
+            }
+        }
+
+        String outputFilename = options.get(KW_OUTPUT_FILE);
+        String outputFormat = options.get(KW_OUTPUT_FORMAT);
+        if (outputFormat == null || outputFormat.isEmpty()) {
+            if (outputFilename == null || outputFilename.isEmpty()) {
+                throw new RuntimeException("[format error] unspecified format of output");
+            } else {
+                int extensionIndex = outputFilename.lastIndexOf(".");
+                if (extensionIndex >= 0) {
+                    outputFormat = outputFilename.substring(extensionIndex + 1);
+                } else {
+                    throw new RuntimeException("[format error] unspecified format of output file");
+                }
+            }
+        }
+
+        Writer writer;
+        if (outputFilename == null || outputFilename.isEmpty()) {
+            writer = new StringWriter();
+        } else {
+            File outputFile = new File(outputFilename);
+            if (!outputFile.exists()) {
+                outputFile.createNewFile();
+            }
+            writer = new FileWriter(outputFile);
+        }
+        try {
+            process(textEngine.toString(), writer, inputFormat, outputFormat, options);
+        } catch (Exception ex) {
+            writer.close();
+            throw ex;
+        }
+        writer.flush();
+        if (outputFilename == null || outputFilename.isEmpty()) {
+            StringWriter stringWriter = (StringWriter) writer;
+            System.out.println(stringWriter.getBuffer().toString());
+        }
+        writer.close();
+    }
+
+    protected static void process(String input, Writer writer,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.